### PR TITLE
code refactoring and picking hardcore values from config

### PR DESCRIPTION
**Pull request by ajinkyaghodake123**

### CODE REVIEW

**Review by waibhavchandra — APPROVED** [review]

**Review by mathewjpallan — CHANGES_REQUESTED** [review]

File: router/appRest.js [hunk 1]
```diff
@@ -100,18 +110,18 @@ function handler(req, res, data, userId, channel ) {
 				setRedisKeyValue(data.customData.deviceId, userData);
 				data.customData.sessionId = uuID;
 				telemetry.logSessionStart(data.customData);
-				if (channel == "botclient") {
-					telemetryData = createInteractionData({ currentStep: 'step1', responseKey: chatflowConfig['step1']['messageKey'] }, data.customData, false, channel)
+				if (data.channel == "botclient") {
+					telemetryData = createInteractionData({ currentStep: 'step1', responseKey: chatflowConfig['step1']['messageKey'] }, data.customData, false, data.channel)
 					telemetry.logInteraction(telemetryData)
-					sendChannelResponse(res, chatflowConfig['step1']['messageKey'], channel, userId);
+					sendChannelResponse(res, chatflowConfig['step1']['messageKey'], data.channel, data.recipient);
```
[the remark this answers is not in the document]
Comment (ajinkyaghodake123) [reply router/appRest.js]: done

File: router/appRest.js [hunk 2]
```diff
@@ -86,10 +96,10 @@ function handler(req, res, data, userId, channel ) {
 				// all non numeric user messages go to bot
 				if (isNaN(data.message)) {
 					///Bot interaction flow
-					freeFlowLogic(data, res, channel, chatflowConfig, userId)
+					freeFlowLogic(data, res, data.channel, chatflowConfig, data.recipient)
 
 				} else {
-					menuDrivenLogic(data, res, channel, chatflowConfig, userId)
+					menuDrivenLogic(data, res, data.channel, chatflowConfig, data.recipient)
```
[the remark this answers is not in the document]
Comment (ajinkyaghodake123) [reply router/appRest.js]: done

File: router/config/config.js [hunk 3]
```diff
@@ -22,14 +22,39 @@ config.TELEMETRY_DATA_PID          = env.telemetry_pid || 'dikshavani.botclient'
...
+config.SECRET_KEY                  = ''
 
 config.REDIS_PORT					 = 6379
 config.REDIS_HOST					 = env.REDIS_HOST || 'localhost'
 config.ELASTIC_HOST                  = 'http://<user>:<password><IP>:<Port>'
 config.ELASTIC_INDEX_NAME            = 'indx_name'
 config.ELASTIC_INDEX_TYPE            = 'indx_type'
 
+config.WHATSAPP_URL                = 'https://waapi.pepipost.com/api/v2/message/'
+config.WHATSAPP_AUTHORIZATION      = 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJuZXRjb3Jlc2FsZXNleHAiLCJleHAiOjI0MjUxMDI1MjZ9.ljC4Tvgz031i6DsKr2ILgCJsc9C_hxdo2Kw8iZp9tsVcCaKbIOXaFoXmpU7Yo7ob4P6fBtNtdNBQv_NSMq_Q8w'
```
[the remark this answers is not in the document]
Comment (ajinkyaghodake123) [reply router/config/config.js:34]: DONE

File: router/appRest.js [hunk 4]
```diff
@@ -347,10 +356,10 @@ function sendChannelResponse(response, responseKey, channel, userId, responseCod
 function createInteractionData(responseData, userData, isNonNumeric, channel) {
 	subtypeVar = ''
 	if (isNonNumeric) {
-		if (channel == 'whatsapp'){
-			subtypeVar = responseData.intent ? 'whtsapp_Free_flow_intent_detected' : 'whtsapp_Free_flow_intent_not_detected'
+		if (channel == config.WHATSAPP){
```
[the remark this answers is not in the document]
Comment (ajinkyaghodake123) [reply router/appRest.js]: @mathewjpallan Keeping the subtype channel specific will  any problem in further queries?

**Review by mathewjpallan — APPROVED** [review]

**Pull request merged**